samcli/lib/providers/provider.py

```diff
@@ -138,6 +161,36 @@ def architecture(self) -> str:
             )
         return str(arch_list[0])
 
+    @property
+    def build_info(self) -> FunctionBuildInfo:
```

Comment:
Nit: Should this be a dynamic property or something we could set once on creation?